sdk/src/client/api/block_builder/input_selection/mod.rs

```diff
@@ -61,57 +76,80 @@ pub struct Selected {
     pub remainders: Vec<RemainderData>,
     /// Mana rewards by input.
     pub mana_rewards: HashMap<OutputId, u64>,
+    /// Required context inputs.
+    pub context_inputs: HashSet<ContextInput>,
+    /// Mana allotments.
+    pub mana_allotments: Vec<ManaAllotment>,
 }
 
 impl InputSelection {
-    fn required_account_nft_addresses(&self, input: &InputSigningData) -> Result<Option<Requirement>, Error> {
-        let required_address = input
-            .output
-            .required_address(self.slot_index, self.protocol_parameters.committable_age_range())?
-            .expect("expiration unlockable outputs already filtered out");
-
-        let required_address = if let Address::Restricted(restricted) = &required_address {
-            restricted.address()
-        } else {
-            &required_address
-        };
-
-        match required_address {
-            Address::Account(account_address) => Ok(Some(Requirement::Account(*account_address.account_id()))),
-            Address::Nft(nft_address) => Ok(Some(Requirement::Nft(*nft_address.nft_id()))),
-            _ => Ok(None),
-        }
-    }
+    /// Creates a new [`InputSelection`].
+    pub fn new(
+        available_inputs: impl IntoIterator<Item = InputSigningData>,
+        context_inputs: impl IntoIterator<Item = ContextInput>,
```

Comment:
I personally wouldn't have it as mandatory parameter but have a `with_ context_inputs`. For the vast majority of scenario this would be empty.

Reply:
fair point!